### DISCUSSION TITLE
How to get Puppeteersharp to work with Asp.net Core .NET 6.0?

### DISCUSSION THREAD

**Discussion by donnyv** [hardkoded/puppeteer-sharp#2264] [Q&A]
I'm trying to call Puppeteersharp from a controller endpoint but it needed AspNetWebSocketTransport. Which I did find and added to my project. But AspNetWebSocketTransport uses HostingEnvironment.QueueBackgroundWorkItem. Which has been removed in in Asp.net Core .NET 6.0. It looks like background tasks need to be registered as a service now during startup. 

Is there an update coming up that fixes this?

**Comment by kblok** [comment]
There are no plans on this side. But it looks like something interesting to see. Shouldn't it be quite easy to implement a [BackgroundService](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-7.0&tabs=visual-studio)?

**Comment by donnyv** [comment]
@kblok I feel like the way Puppeteersharp is currently handling socket transport would have to change to use BackgroundService. But I haven't done a deep dive on it. So I could be wrong.

**Comment by kblok** [comment]
@donnyv you can pass a [TransportFactory](https://github.com/hardkoded/puppeteer-sharp/blob/master/lib/PuppeteerSharp/Transport/TransportFactory.cs#L14) into the [LaunchOptions](https://github.com/hardkoded/puppeteer-sharp/blob/master/lib/PuppeteerSharp/LaunchOptions.cs#L146).
We have [an example of an ASP.NET Framework transport factory](https://github.com/hardkoded/puppeteer-sharp/blob/master/lib/PuppeteerSharp.AspNetFramework/AspNetWebSocketTransport.cs#L10).
